### PR TITLE
Module aliasing: Resolve deps graph with real module names when scanning dependencies

### PR DESCRIPTION
**Pull request by elsh**
Resolved dependencies should contain the real module name if module aliasing is used. 
Resolves rdar://85991587.

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by artemcm — APPROVED** [review]

File: test/Frontend/module-alias-scan-deps.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/// Scaned dependencies with implicit builds should contain real name AppleLogging
+// RUN: %target-swift-frontend -scan-dependencies  %t/FileLib.swift -module-alias XLogging=AppleLogging -I %t > %t/scandump.output
+// RUN: %FileCheck %s -check-prefix=CHECK-REAL-NAME -input-file  %t/scandump.output
+// CHECK-REAL-NAME-NOT: "swiftPrebuiltExternal": "XLogging"
+// CHECK-REAL-NAME-NOT: "compiledModulePath":{{.*}}XLogging.swiftmodule",
+// CHECK-REAL-NAME: "swiftPrebuiltExternal": "AppleLogging"
+// CHECK-REAL-NAME: "compiledModulePath":{{.*}}AppleLogging.swiftmodule",
+
+/// Scaned dependencies with explicit builds should contain real name AppleLogging
+// RUN: %target-swift-frontend -scan-dependencies -disable-implicit-swift-modules %t/FileLib.swift -module-alias XLogging=AppleLogging -I %t > %t/scandumpExplicit.output
```
Comment (artemcm) [inline test/Frontend/module-alias-scan-deps.swift:22]: `-disable-implicit-swift-modules` is not necessary here, it should not have an effect during scanning.

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by beccadax** [comment]
Retrospective review: LGTM.

**Pull request merged**